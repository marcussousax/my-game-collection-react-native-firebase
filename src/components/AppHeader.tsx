import React from 'react'
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native'

import { AuthContext } from '../contexts/auth'

// To be used with Authenticated User
const AppHeader: React.FC<{
    title?: string
    hideAvatar?: boolean
    navigation?: any
}> = ({ title, hideAvatar, navigation }) => {
    const { user } = React.useContext(AuthContext)

    return (
        <View style={styles.header}>
            {title ? <Text style={styles.title}>{title}</Text> : null}
            {!hideAvatar ? (
                <View>
                    <TouchableOpacity
                        onPress={() => navigation.navigate('ProfileScreen')}
                    >
                        <Image
                            style={styles.tinyLogo}
                            source={{ uri: user?.photoURL || undefined }}
                        />
                    </TouchableOpacity>
                </View>
            ) : null}
        </View>
    )
}

const styles = StyleSheet.create({
    header: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingVertical: 20
    },
    title: {
        color: '#02cbd3',
        fontSize: 20,
        fontWeight: 'bold'
    },
    tinyLogo: {
        width: 32,
        height: 32,
        borderRadius: 20
    }
})

export default AppHeader
