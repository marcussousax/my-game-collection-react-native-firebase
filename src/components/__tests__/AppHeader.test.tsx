import 'react-native'
import React from 'react'
import AppHeader from '../AppHeader'
import TestRenderer from 'react-test-renderer'

describe('<AppHeader>', () => {
    it('it renders correctly', () => {
        const header = TestRenderer.create(
            <AppHeader title="my game collection" />
        ).toJSON()
        expect(header).toMatchSnapshot()
    })

    it('it shows the title', () => {
        const header = TestRenderer.create(
            <AppHeader title="my game collection" />
        )
        const componentInstance = header.root.findByType(AppHeader)
        expect(componentInstance.props.title).toBe('my game collection')
    })
})
