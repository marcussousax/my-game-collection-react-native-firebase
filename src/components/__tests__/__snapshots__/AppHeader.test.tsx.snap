// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader> it renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#02cbd3",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    my game collection
  </Text>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 20,
            "height": 32,
            "width": 32,
          }
        }
      />
    </View>
  </View>
</View>
`;
